frontend/src/App.js

```diff
@@ -4,31 +4,36 @@ import Navbar from './components/navbar/Navbar';
 import Home from './pages/homepage/Home';
 import Signup from './pages/signup/Signup';
 import Bottombar from './components/bottombar/Bottombar';
-import {BrowserRouter,Routes,Route} from "react-router-dom"
-import {FilterProvider} from "./context/Filtercontext"
-import {AuthProvider} from "./context/AuthContext"
+import { BrowserRouter, Routes, Route } from "react-router-dom"
+import { FilterProvider } from "./context/Filtercontext"
+import { AuthProvider } from "./context/AuthContext"
 import Login from './pages/login/Login';
+import Cart from './pages/cartpage/Cart';
 import { ProductProvider } from './context/ProductContext';
+import { CartProvider } from './context/CartContext';
 function App() {
   return (
     <div className="App">
-    
-    <BrowserRouter>
-    <AuthProvider>
-    <ProductProvider>
-    <FilterProvider>
-    <Navbar/>
-    <Routes>
-    <Route path="/" element={<Home/>}/>
-    <Route path="/product" element={<ProductPage/>}/>
-    <Route path="/signup" element={<Signup/>}/>
-    <Route path='/login' element={<Login/>}/>
-    </Routes>
-    <Bottombar/>
-    </FilterProvider>
-    </ProductProvider>
-    </AuthProvider>
-    </BrowserRouter>
+
+      <BrowserRouter>
+        <AuthProvider>
+          <ProductProvider>
+            <CartProvider>
+              <FilterProvider>
+                <Navbar />
+                <Routes>
+                  <Route path="/" element={<Home />} />
+                  <Route path="/product" element={<ProductPage />} />
+                  <Route path="/signup" element={<Signup />} />
+                  <Route path='/login' element={<Login />} />
+                  <Route path='/cart' element={<Cart />} />
+                </Routes>
+                <Bottombar />
+              </FilterProvider>
+            </CartProvider>
+          </ProductProvider>
+        </AuthProvider>
+      </BrowserRouter>
```

Comment:
suggestion: you can put the providers in `index.js`

Reply:
Thanks for the suggestion @iRohitGaur ,I will shift those providers to index.js.